power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -328,11 +329,16 @@ inline auto get_edge_weights(TransformerGraph const& graph) -> TrafoGraphEdgePro
         // side via the bidirectional edge (if it exists). For delta configuration ABC, the above
         // situations can happen.
         // The logic still holds in meshed grids, albeit operating a more complex graph.
-        if (edge_src_rank != edge_tgt_rank - 1) {
-            throw AutomaticTapInputError("The control side of a transformer regulator should be relatively further "
-                                         "away from the source than the tap side.\n");
-        }
         if (!is_unreachable(edge_res)) {
+            if (edge_src_rank == infty && edge_tgt_rank != infty) {
+                throw AutomaticTapInputError("The transformer is being controlled from non source side towards source "
+                                             "side.\n");
+            }
```

Comment:
Can we simply only do the following? The rest in this block will remain put
```suggestion
            if (edge_tgt_rank == 0) {
                throw AutomaticTapInputError("The transformer control side is directly connected to source.\n");
            }
```

Reply:
Resolving as we have discussed this offline.